etc/update_version/src/main.rs

```diff
@@ -74,45 +74,54 @@ struct Args {
 }
 
 fn main() {
+    // nosemgrep: current-exe
+    let zero = std::env::current_exe().unwrap();
+    let self_dir = zero.parent().unwrap();
+    let main_dir = self_dir.join("../../../..");
+    std::env::set_current_dir(main_dir).unwrap();
+
     let version_locs = vec![
         Location::new(
-            "../../Cargo.toml",
+            "Cargo.toml",
             r#"name = "mongodb"\nversion = "(?<target>.*?)"\n"#,
         ),
-        Location::new("../../README.md", r#"mongodb = "(?<target>.*?)"\n"#),
         Location::new(
-            "../../README.md",
-            r#"\[dependencies.mongodb\]\nversion = "(?<target>.*?)"\n"#,
+            "macros/Cargo.toml",
+            r#"name = "mongodb-internal-macros"\nversion = "(?<target>.*?)"\n"#,
         ),
-        Location::new("../../src/lib.rs", r#"//! mongodb = "(?<target>.*?)"\n"#),
-        Location::new("../../src/lib.rs", r#"//! version = "(?<target>.*?)"\n"#),
+        Location::new("README.md", r#"mongodb = "(?<target>.*?)"\n"#),
         Location::new(
-            "../../src/lib.rs",
-            r#"html_root_url = "https://docs.rs/mongodb/(?<target>.*?)""#,
+            "README.md",
+            r#"\[dependencies.mongodb\]\nversion = "(?<target>.*?)"\n"#,
         ),
         Location::new(
-            "../../manual/src/installation_features.md",
-            r#"\[dependencies.mongodb\]\nversion = "(?<target>.*?)"\n"#,
+            "src/lib.rs",
+            r#"html_root_url = "https://docs.rs/mongodb/(?<target>.*?)""#,
         ),
     ];
-    let bson_version_loc =
-        Location::new("../../Cargo.toml", r#"bson = (?<target>\{ git = .*? \})\n"#);
-    let mongocrypt_version_loc = Location::new(
-        "../../Cargo.toml",
-        r#"mongocrypt = (?<target>\{ git = .*? \})\n"#,
+    let quote_version_loc = Location::new(
+        "Cargo.toml",
+        r#"mongodb-internal-macros = (?<target>\{ path = .* \})\n"#,
```

Comment:
I expect the regex will no longer match once `mongodb-internal-macros` is added to Cargo. Suggest adding a comment to remind:
```suggestion
        // TODO: after `mongodb-internal-macros` is added to Cargo, update regex to:
        // r#"mongodb-internal-macros = (?<target>.*?)\n"#
        r#"mongodb-internal-macros = (?<target>\{ path = .* \})\n"#,
```

Reply:
I don't think this will be needed - `mongodb-internal-macros` will only be updated to point to crates.io for releases, on the dev branch it'll always be a `path` dependency.  We treat `bson` similarly, except with a `git` dependency rather than `path` since it's not in the same repo.